Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -85,3 +85,22 @@ public class OpenFeatureAPI {
         .eraseToAnyPublisher()
     }
 }
+
+extension OpenFeatureAPI {
+    public func setProviderAndWait(provider: FeatureProvider) async {
+        await setProviderAndWait(provider: provider, initialContext: nil)
+    }
+
+    public func setProviderAndWait(provider: FeatureProvider, initialContext: EvaluationContext?) async {
+        var holder: [AnyCancellable] = []
+        await withCheckedContinuation { continuation in
+            let stateObserver = provider.observe().sink {
+                if $0 == .ready {
+                    continuation.resume()
+                }
+            }
+            stateObserver.store(in: &holder)
```

Comment:
This is used to keep the Cancellable from `sink` alive until we receive `.ready`. We are not 100% that holding this in a local property inside this function is the best pattern

Reply:
can we `dispose` or something similar on holder after the resume? and also if there's any callback for cancellation?

Reply:
Can do probably, although my reasoning is that without retain cycles in this function all these objects should be released automatically upon return

Reply:
is there a function to see the number of subscribers for an observable?

Reply:
i mean, we can assert the number of subscribers in the test and should be 0 after the function returning.

Reply:
Let's resolve this thread and make sure to open new comments if testing for this PR is lacking in some areas